### ISSUE TITLE
Re: Not able to generate features properly using ADAP workflow in MZmine 2

### ISSUE THREAD

**Issue by JianFelixGuo** [mzmine/mzmine2#963]
### Basic information

* My operating system: windows 10
* My MZmine version: 2.53

### What happened
I was trying to follow the workflow provided in the user manual to process one mzXML file using MZmine 2.  The following is my parameter setup.
step1: import the raw data
I used a urine data file which should contain several thousands features normally.
step2: mass detection
![image](https://user-images.githubusercontent.com/101300794/161648130-c929c543-f47b-4841-86da-7a7b7b785abe.png)
step3 : ADAP chromatogram builder
![image](https://user-images.githubusercontent.com/101300794/161648315-a28a4fa8-08e7-47aa-943b-e01b5cf1135c.png)
step4 : ADAP chromatogram deconvolution
![image](https://user-images.githubusercontent.com/101300794/161648438-e612a530-a97b-45e0-9b15-fe3c015a4d04.png)
However, I could only get 7 features in the final feature table!
![image](https://user-images.githubusercontent.com/101300794/161648602-236acf85-1d33-494e-a7b0-9ae701b13511.png)

I also tested several other data files, and I can always get less than 100 features when I could get thousands of features from MS-DIAL. Is there anything wrong with the parameters or the workflow I used? Many thanks if you could enlighten me on why this is happening.

**Comment by asmirn1** [comment]
@JianFelixGuo 

Your detected features seem to have low intensities. I assume that the undetected features are even lower. So, first I'd try to decrease the Noise Level, Group Intensity, Min Highest Intensity, and Min Feature Height thresholds. Maybe, try these values:
- Mass Detection: Noise Level = 0
- Chromatogram Builder: Group Intensity Threshold = 50
- Chromatogram Builder: Min Highest Intensity = 100
- Chromatogram Deconvolution: Min Feature Height = 100
- Chromatogram Deconvolution: Coefficient/Area threshold = 0

Another parameter that you can try to adjust is the RT Wavelet Range. The value 0.48 seems very high. It might detect long peaks (maybe ~1 min wide), but will skip many shorter peaks. It's hard to tell what is the optimum parameter here without looking at the data. But I'd suggest trying values 0.05, 0.1, etc. and see how the results are changing.

One very useful feature of MZmine is the Preview. Please use the preview to see immediate results.

**Comment by JianFelixGuo** [comment]
@asmirn1 Hi Alex, 

Thanks a lot for the instant reply! Regarding your two suggestions, I have some results to show you. 
1. The intensity should not be the problem. First, below is the chromatogram of the raw data I used (urine ran in RP+ mode with Bruker Q-Tof). You can see there are many high-abundant peaks in the BPC.
![image](https://user-images.githubusercontent.com/101300794/161816559-ed38ea9e-e3a4-41e9-82a0-db09fdec2760.png)
Plus, I used identical parameters (same mass tolerance, noise level, group intensity threshold, S/N threshold, and RT wavelet range as the ones used in ADAP) in XCMS centwave to process the same data to test their different performances because CentWave also uses Wavelet fitting similar to ADAP's algorithm and is thus comparable. Here attached is the feature table I got from XCMS. You can see many high-abundant features are there. 
![image](https://user-images.githubusercontent.com/101300794/161817074-f6c22fae-0c22-45d2-8abe-6824568814dd.png)
Generally, the noise level of our data generated by Bruker Q-TOF is around 300-500. If the noise level is too low, there would be lots of the false positive features.
![image](https://user-images.githubusercontent.com/101300794/161821954-82dec3aa-95fc-4b47-b979-41fecbe6de7e.png)

 2. As for the RT wavelet range, I followed your advice as shown below. The lower range is set to be zero and upper limit is set to be half of the maximum peak width in my data.
![image](https://user-images.githubusercontent.com/101300794/161818421-aec0dc63-3879-479e-a0af-bca78e21d9cd.png)
However, I do agree that the RT wavelet range might be the issue. I did try some lower values like 0.05 and 0.1. Suprisingly, I got no feature generated when using 0.05, and only 12 features when using 0.1!
![image](https://user-images.githubusercontent.com/101300794/161820058-f3c8a73b-d89d-4b73-a93b-a340f1b2d4bb.png)
I wonder if the peak shape in the RP+ data is too narrow and ADAP can not recognize them, so I tried it on my HILIC- urine data. I could get a bit more features but still way lower than I expected.
![image](https://user-images.githubusercontent.com/101300794/161820546-7ab307a7-a049-4bb2-aba1-35a91ef8e081.png)
I also noticed all the deteced features are in the low intensity range. I wonder if a different chromatogram deconvolution algorithm might work better than ADAP wavelet. So I tried the local minimum search (default method), and I could get a decent amount of features now (like 6263 features from RP+ urine data). 
![image](https://user-images.githubusercontent.com/101300794/161821073-1ff19b42-0f2f-49c1-9f29-c814f381d668.png)
Do you recommed using local minimum search instead of ADAP wavelet when such things happen? I have tried several more datasets and consistently got more features using local minimum search than ADAP wavelet.

**Comment by asmirn1** [comment]
You are right. It looks like the ADAP Chromatogram Deconvolution is an issue here.

Personally, I don't have much experience with the local minimum search. But if it works for you, that's great.

I wonder if you could share one of your data files or your MZmine project file with me. Maybe I could see what's wrong with the ADAP Wavelet parameters.

Alex

**Comment by JianFelixGuo** [comment]
@asmirn1 Thanks Alex, I only used one mzXML file for the data processing, so I could simply send that file to you. What is the best email address I can send this file to?

**Comment by JianFelixGuo** [comment]
@asmirn1 do you have any suggestions on the parameters used for local minimum search? Such as the chromatogram threshold and minim top/edge ratio. These two parameters seem to influence the results a lot and very subjective at the same time.

**Comment by ibokaraman** [comment]
Hi @JianFelixGuo,
This has drawn my attention as I use ADAP methods.
How many peak lists do you get after the chromatogram building step? Looking at your settings, must be many.
Then could you share a screenshot of a preview of a feature (e.g. the one with highest intensity after chromatogram building) during chromatogram deconvolution step?
Thanks
Ibrahim

**Comment by JianFelixGuo** [comment]
Hi @ibokaraman,
Thanks for joining the discussion. There were 5559 chromatograms before deconvolution.
![image](https://user-images.githubusercontent.com/101300794/162794708-27c7e04c-0fc3-4c0c-9762-4688e9d26005.png)
After ADAP deconvolution, there were only 12 features. Below is a screenshot of the feature with highest intensity after deconvolution.
![image](https://user-images.githubusercontent.com/101300794/162794944-3b0355f7-45fa-40b7-a0ca-2e1d12db0362.png)
